pkg/tfbridge/schema.go

```diff
@@ -501,7 +501,22 @@ func MakeTerraformResult(state *terraform.InstanceState,
 			Schema: tfs,
 			Map:    schema.BasicMapReader(attrs),
 		}
-		for _, key := range flatmap.Map(attrs).Keys() {
+
+		// Read each top-level field out of the attributes.
+		keys := make(map[string]bool)
+		for key := range attrs {
+			// Pull the top-level field out of this attribute key. If we've already read the top-level field, skip this
+			// key.
+			dot := strings.Index(key, ".")
```

Comment:
Is there only ever one `.` in these attrs?

Reply:
No: there is frequently more than one `.`. We are specifically looking for the _first_ dot, as we only need the component of the attribute key that precedes that first dot. That portion of the key is the name of the top-level property in which attribute is contained.